File: examples/typescript/vectorize-file.ts
import { readFile, writeFile } from "node:fs/promises";
import { Configuration, VectorizationApi } from "@vectorizer-ai/sdk";

const apiId = process.env.VECTORIZER_API_ID;
const apiSecret = process.env.VECTORIZER_API_SECRET;
const inputPath = process.env.VECTORIZER_INPUT ?? "example.png";
const outputPath = process.env.VECTORIZER_OUTPUT ?? "result.svg";

if (!apiId || !apiSecret) {
  throw new Error("Set VECTORIZER_API_ID and VECTORIZER_API_SECRET.");
}

const api = new VectorizationApi(
  new Configuration({
    username: apiId,
    password: apiSecret,
  }),
);

const image = new Blob([await readFile(inputPath)]);
const result = await api.postVectorize({
  image,
  outputFileFormat: "svg",
});

await writeFile(outputPath, Buffer.from(await result.arrayBuffer()));
console.log(`Wrote ${outputPath}`);

File: examples/csharp/VectorizeFile.cs
using Vectorizer.AI.Api;
using Vectorizer.AI.Client;

var apiId = Environment.GetEnvironmentVariable("VECTORIZER_API_ID");
var apiSecret = Environment.GetEnvironmentVariable("VECTORIZER_API_SECRET");
var inputPath = Environment.GetEnvironmentVariable("VECTORIZER_INPUT") ?? "example.png";
var outputPath = Environment.GetEnvironmentVariable("VECTORIZER_OUTPUT") ?? "result.svg";

var config = new Configuration
{
    Username = apiId,
    Password = apiSecret,
};

var api = new VectorizationApi(config);
using var image = File.OpenRead(inputPath);
using var result = api.PostVectorize(
    image: image,
    outputFileFormat: "svg");

using var output = File.Create(outputPath);
result.CopyTo(output);
Console.WriteLine($"Wrote {outputPath}");
